### PR TITLE
return an LDAPSearchResultEntry even if all attributes are filtered

### PR DESCRIPTION
**Pull request by graingert**
## Remove this paragraph

Please have a look at our dev docs before submitting your PR:
https://github.com/twisted/ldaptor/blob/master/CONTRIBUTING.rst

### Contributor Checklist:

* [ ] I have updated the release notes at `docs/source/NEWS.rst` 
* [ ] I have updated the automated tests.
* [ ] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

**Comment by graingert** [comment]
This is a fix to:

https://github.com/twisted/ldaptor/commit/b96df7543c4d491ba1d7c7624bc7ae9d4ec9118c

https://github.com/twisted/ldaptor/pull/40

when running the same search on MS AD with empty attrs the object DN is returned with no attributes

**Comment by graingert** [comment]
@adiroiban

File: docs/source/NEWS.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ Bugfixes
 ^^^^^^^^
 
 - SASL Bind without credentials caused list index out of range. Issue #157, Fixed
+- return an LDAPSearchResultEntry even if all attributes are filtered. Issue #166, Fixed
```
[the remark this answers is not in the document]
Comment (graingert) [reply docs/source/NEWS.rst]: Can you make this a GitHub suggestion?

**Comment by graingert** [comment]
I think this PR implicitly adds support for "1.1" behaviour

**Comment by adiroiban** [comment]
> I think this PR implicitly adds support for "1.1" behaviour

True.

> Can you make this a GitHub suggestion?

I don't know how to add a commit suggestion.
Is this feature available to free accounts?

But I have committed my suggestion.

Please check the changes and let me know if the description is correct.

Thanks!

We can then merge it if all tests are green... with the exception of the test coverage

**Comment by graingert** [comment]
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request step 6

File: docs/source/NEWS.rst [hunk 2]
```diff
@@ -13,10 +13,10 @@ Changes
...
 ^^^^^^^^
 
-- SASL Bind without credentials caused list index out of range. Issue #157, Fixed
-- return an LDAPSearchResultEntry even if all attributes are filtered. Issue #166, Fixed
-
-
+- SASL Bind without credentials caused list index out of range. Issue #157.
+- ldaptor.protocols.ldap.ldapserver.LDAPServer.handle_LDAPSearchRequest
+  now returns an LDAPSearchResultEntry without any attributes when there is no match
+  between the requested attributes and the entrie's attributes .Issue #166.
```
Comment (graingert) [inline docs/source/NEWS.rst]:
```suggestion
  between the requested attributes and the entrie's attributes. Issue #166.
```

**Pull request merged**